apm-agent-core/src/main/java/co/elastic/apm/agent/tracemethods/TraceMethodInstrumentation.java

```diff
@@ -153,7 +117,55 @@ public boolean matches(MethodDescription target) {
     }
 
     @Override
-    public boolean indyPlugin() {
-        return false;
+    public String getAdviceClassName() {
+        return getClass().getName() + "$TraceMethodAdvice";
+    }
+
+    public static class TraceMethodAdvice {
+
+        private static final ElasticApmTracer tracer = GlobalTracer.requireTracerImpl();
```

Comment:
[minor] moving tracer init to the start of the `static` block makes the dependency to retrieve configuration obvious.